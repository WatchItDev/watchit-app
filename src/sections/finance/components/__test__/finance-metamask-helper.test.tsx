import { describe, expect, it } from 'vitest';
import { render, screen } from '@testing-library/react';
import FinanceMetamaskHelper from '../finance-metamask-helper';

describe('<COMPONENTS> FinanceMetamaskHelper', () => {
  it('to match snapshot', () => {
    const { container } = render(<FinanceMetamaskHelper />);
    expect(container).toMatchSnapshot();
  });

  it('renders Stack, Typography, and Link components', () => {
    render(<FinanceMetamaskHelper />);
    expect(screen.getByText(/MetaMask required/i)).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: /Click here/i }),
    ).toBeInTheDocument();
  });

  it('Link has correct attributes', () => {
    render(<FinanceMetamaskHelper />);
    const link = screen.getByRole('link', { name: /Click here/i });
    expect(link).toHaveAttribute('href', 'https://metamask.app.link');
    expect(link).toHaveAttribute('target', '_blank');
    expect(link).toHaveAttribute('rel', 'noopener');
  });
  it('Link has correct styles', () => {
    render(<FinanceMetamaskHelper />);
    const link = screen.getByRole('link', { name: /Click here/i });
    expect(link).toHaveStyle('color: rgba(0, 0, 0, 0.87);');
  });

  it('calls handleDownloadMetaMask on click', () => {
    render(<FinanceMetamaskHelper />);
    const link = screen.getByRole('link', { name: /Click here/i });
    link.click();
    expect(window.location.href).toBe('http://localhost:3000/');
  });
});
