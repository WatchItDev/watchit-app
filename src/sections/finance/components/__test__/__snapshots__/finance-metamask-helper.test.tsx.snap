// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<COMPONENTS> FinanceMetamaskHelper > to match snapshot 1`] = `
<div>
  <div
    class="MuiStack-root css-1i3xkqc-MuiStack-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-1da0jfs-MuiTypography-root"
    >
      MetaMask required.
       
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1pdnr0o-MuiTypography-root-MuiLink-root"
        href="https://metamask.app.link"
        rel="noopener"
        target="_blank"
      >
        Click here
      </a>
       
      if you don't have it installed.
    </p>
  </div>
</div>
`;
